### PR TITLE
Add support for phpcr odm

### PR DESCRIPTION
**Pull request by alekitto**
As per subject, adds document iterator and repository for phpcr odm

### CODE REVIEW

File: src/DBAL/TestStatement.php [hunk 1]
```diff
@@ -0,0 +1,206 @@
+<?php declare(strict_types=1);
+
+namespace Fazland\DoctrineExtra\DBAL;
+
+use Doctrine\DBAL\Driver\Statement;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\ParameterType;
+
+class TestStatement implements \IteratorAggregate, Statement
```
Comment (massimilianobraglia) [inline src/DBAL/TestStatement.php]: Is this class necessary into the `Fazland\DoctrineExtra\DBAL` namespace?
I imagine that you wrote this for test purposes, am I wrong?

File: src/DBAL/TestStatement.php [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+{
+    /**
+     * @var mixed[]
+     */
+    private $data;
+
+    /**
+     * @var int
+     */
+    private $columnCount = 0;
```
Comment (massimilianobraglia) [inline src/DBAL/TestStatement.php]: Could this properties be initialized in the constructor?

File: src/DBAL/TestStatement.php [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+    /**
+     * @var int
+     */
+    private $num = 0;
```
Comment (massimilianobraglia) [inline src/DBAL/TestStatement.php]: as above

**Review by massimilianobraglia — COMMENTED** [review]
Hey @alekitto! Great job, just left some minor comments

File: src/DBAL/TestStatement.php (hunk 1 above)
Comment (alekitto) [reply src/DBAL/TestStatement.php]: For test purposes yes, but re-usable from others libs/projects.

File: src/DBAL/TestStatement.php (hunk 2 above)
Comment (alekitto) [reply src/DBAL/TestStatement.php]: Fixed

File: src/DBAL/TestStatement.php (hunk 3 above)
Comment (alekitto) [reply src/DBAL/TestStatement.php]: Fixed

**Comment by massimilianobraglia** [comment]
Thank you, @alekitto!

**Pull request merged**